### ISSUE TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'getBoundingClientRect')

### ISSUE THREAD

**Issue by george-seed** [ANovokmet/svelte-gantt#176]
I'm using SvelteKit and I'm finding that random items in the gantt chart are becoming un-selectable. Specifically, it sems to happen whenever I navigate to another route, then back again.

It appears to be an issue with the domUtils.ts script but I'm not familiar enough with the library to diagnose.

Uncaught TypeError: Cannot read properties of undefined (reading 'getBoundingClientRect')
    at getRelativePos (domUtils.ts:11:23)
    at Draggable.onmousedown (draggable.ts:100:35)

```
export function getRelativePos(node: HTMLElement, event: MouseEvent) {
    const rect = node.getBoundingClientRect();
    const x = event.clientX - rect.left; //x position within the element.
    const y = event.clientY - rect.top;  //y position within the element.
    return {
        x: x,
        y: y
    }
}
```

I thought I'd report it. Here's my script for reference.

```
<script>
    import { SvelteGantt, SvelteGanttDependencies, SvelteGanttExternal, SvelteGanttTable, MomentSvelteGanttDateAdapter} from 'svelte-gantt';
    import { onMount } from 'svelte';
    import moment from 'moment';

    onMount(() => {
        function time(input) {
        return moment(input, 'HH:mm');
        }

    const currentStart = time('06:00');
    const currentEnd = time('18:00');

    const colors = ['blue', 'green', 'orange','red']

        const timeRanges = [
        {
            id: 0,
            from: time('10:00'),
            to: time('12:00'),
            classes: null,
            label: 'Lunch'
        },
        {
            id: 1,
            from: time('15:00'),
            to: time('17:00'),
            classes: null,
            label: 'Dinner'
        }
    ];

    const data = {
        rows: [{
            "id": 1,
            "label": "Preparation and Planning"
        }, {
            "id": 2,
            "label": "Development"
        }, {
            "id": 3,
            "label": "Implementation"
        }, {
            "id": 4,
            "label": "Training"
        }, {
            "id": 5,
            "label": "Roll-out product"
        }],
        tasks: [{
            "id": 1,
            "resourceId": 1,
            "label": "Preparation",
            "from": time("7:00"),
            "to": time("9:00"),
            "classes": "orange"
        }, {
            "id": 2,
            "resourceId": 2,
            "label": "Planning",
            "from": time("6:30"),
            "to": time("09:00"),
            "classes": "orange"
        }, {
            "id": 3,
            "resourceId": 3,
            "label": "Development",
            "from": time("12:00"),
            "to": time("13:30"),
            "classes": "orange"
        }, {
            "id": 4,
            "resourceId": 4,
            "label": "Implementation",
            "from": time("13:45"),
            "to": time("15:45"),
            "classes": "orange"
        }, {
            "id": 5,
            "resourceId": 5,
            "label": "Finish",
            "from": time("17:00"),
            "to": time("17:45"),
            "classes": "green"
        }, {
            "id": 6,
            "resourceId": 4,
            "label": "Training",
            "from": time("7:00"),
            "to": time("10:00"),
            "classes": "blue"
        }]
    };

    const options = {
        dateAdapter: new MomentSvelteGanttDateAdapter(moment),
        rows: data.rows,
        tasks: data.tasks,
        dependencies: data.dependencies,
        timeRanges,
        columnOffset: 15,
        magnetOffset: 15,
        rowHeight: 52,
        rowPadding: 6,
        headers: [{ unit: 'day', format: 'MMMM Do' }, { unit: 'hour', format: 'H:mm' }],
        fitWidth: true,
        minWidth: 800,
        from: currentStart,
        to: currentEnd,
        tableHeaders: [{ title: 'Resource', property: 'label', width: 140, type: 'tree' }],
        tableWidth: 240,
        ganttTableModules: [SvelteGanttTable],
        ganttBodyModules: []
    }

    let gantt = new SvelteGantt({ target: document.getElementById('example-gantt'), props: options })

});
</script>

    <div>
      <div id="example-gantt">
      </div>
    </div>
```

**Comment by ANovokmet** [comment]
Thank you, looking into it.

**Comment by ANovokmet** [comment]
Fixed in soon to be released v4.1, thank you @george-seed
